### PR TITLE
Add v1.2.1 upgrade docs

### PR DESCRIPTION
**Pull request by bk201**
Add v1.2.1 upgrade docs:
- Add v1.2.1 pages.
- Add notes describing some v1.2.0 known issues that are already fixed in v1.2.1.

### CODE REVIEW

File: docs/upgrade/v1-2-0-to-v1-2-1.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+
+## Important changes to this version
+
+Harvester v1.2.1 fixes upgrade known issues found in v1.2.0, we suggest upgrading v1.1.x clusters to v1.2.1. The known issues are:
+- [An Upgrade is stuck in the Post-draining state](./v1-1-2-to-v1-2-1.md#9-an-upgrade-is-stuck-in-the-post-draining-state)
+- [An upgrade is stuck in the Upgrading System Service state due to the customer provided SSL certificate without IP SAN error in fleet-agent](./v1-1-2-to-v1-2-1.md#10-an-upgrade-is-stuck-in-the-upgrading-system-service-state-due-to-the-customer-provided-ssl-certificate-without-ip-san-error-in-fleet-agent)
+
+
+For clusters that are already upgraded to v1.2.0, please refer to the [release note](https://github.com/harvester/harvester/releases/tag/v1.2.1) for new changes.
```
Comment (Vicente-Cheng) [inline docs/upgrade/v1-2-0-to-v1-2-1.md:19]:
```suggestion
For clusters already upgraded to v1.2.0, please refer to the [release note](https://github.com/harvester/harvester/releases/tag/v1.2.1) for new changes.
```

**Review by Vicente-Cheng — APPROVED** [review]
just a nit, others lgtm. Thanks!

**Comment by ibrokethecloud** [comment]
lgtm. not sure if we want to change the recommended upgrade for v1.1.2, and avoid v1.2.0 as target.

image

File: docs/upgrade/v1-2-0-to-v1-2-1.md (hunk 1 above)
Comment (bk201) [reply docs/upgrade/v1-2-0-to-v1-2-1.md]: Thanks, fixed!

**Review by ibrokethecloud — APPROVED** [review]
LGTM. thanks.

**Pull request merged**